### PR TITLE
Improve VCA (Version/Capabilities/Algorithms) unit test coverage

### PR DESCRIPTION
**Pull request by LeonLinAtAMI**
This PR adds unit tests that slightly improve the overall test coverage rate.
Differences between main and my local coverage report are shown in the attached image.
image

I have prior experience writing thousands of unit tests using xUnit frameworks across multiple languages.
There are various testing styles and rich, diverse testing cultures worldwide. For instance, the Gherkin style:
GIVEN a context, WHEN an action is performed, THEN the expected result should occur.

Please feel free to let me know if:
- any test assertions are still insufficient,
- the testing conventions are incorrect, or
- you have a strong preference for adding more tests to cover specific critical paths.

**Work summary:**

Filled 10 previously-empty test case scaffolds targeting error and edge-case branches in the core Version/Capabilities/Algorithms (VCA) exchange.

Responder GET_VERSION (library/spdm_responder_lib/libspdm_rsp_version.c):
- case2: request smaller than minimum size -> INVALID_REQUEST
- case7: GET_VERSION received with a valid session ID -> UNEXPECTED_REQUEST

Responder GET_CAPABILITIES (library/spdm_responder_lib/libspdm_rsp_capabilities.c):
- case10: unsupported SPDM version -> VERSION_MISMATCH
- case11: request too small for v1.2 -> INVALID_REQUEST
- case12: request too small for v1.1 -> INVALID_REQUEST
- case21: transcript (buffer A) continuity/append verification

Requester NEGOTIATE_ALGORITHMS (library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c):
- case4: responder ERROR InvalidRequest -> LIBSPDM_STATUS_ERROR_PEER
- case5: responder ERROR Busy, no retries -> LIBSPDM_STATUS_BUSY_PEER
- case7: responder ERROR RequestResynch -> LIBSPDM_STATUS_RESYNCH_PEER
- case10: responder selects measurement_hash_algo=0 -> LIBSPDM_STATUS_NEGOTIATION_FAIL

Assisted-by: GitHub Copilot CLI 1.0.78:Claude Sonnet 5

Thanks & best regards,
Leon Lin

### CODE REVIEW

**Comment by LeonLinAtAMI** [comment]
Test coverage updated from my local coverage report.
image

Except unit tests, one line fixed that releasing sender buffer before returning status code.  
image

**Work summary:**
    
Fill scaffold tests and add new cases across the 6 SPDM VCA test files to 
  push line coverage as close to 100% as possible:
- test_spdm_requester/get_version.c + test_spdm_responder/version.c
  - 85.2% -> 98.2%, 94.1% -> 100%
- test_spdm_requester/get_capabilities.c + test_spdm_responder/capabilities.c
 - 87.7% -> 96.7%, 84.7% -> 99.6%
- test_spdm_requester/negotiate_algorithms.c + test_spdm_responder/algorithms.c
 - 87.4% -> 87.8%, 83.9% -> 97.1%
    
New cases cover version-tier size validation, request/response flag
  compatibility failures, SPDM 1.3/1.4 EP_INFO/MULTI_KEY/SET_KEY_PAIR_RESET
  paths, PQC/KEM/ext_flags handling, transcript append failures, busy-retry
  and buffer-too-small paths, and error-code sweeps.
    
Also fixes a genuine bug in libspdm_req_negotiate_algorithms.c: the
  default case for MULTI_KEY_CAP validation returned
  LIBSPDM_STATUS_INVALID_MSG_FIELD without releasing the sender buffer.
    
Assisted-by: GitHub Copilot CLI 1.0.78:Claude Sonnet 5

Thanks & best regards,
Leon Lin

File: library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c [hunk 1]
```diff
@@ -181,6 +181,7 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
             }
             break;
         default:
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:183]: This default branch can be removed entirely. libspdm checks that the Responder has legal `MULTI_KEY_CAP` values when evaluating the `CAPABILITIES` response.

Comment (LeonLinAtAMI) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:183]: To avoid misunderstanding, I shared what I saw in the curent code base from my branch.

Assume 2-bit 11b taken that
    spdm_context->connection_info.capability.flags &
                SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MULTI_KEY_CAP
 was equal to 11b and then went into default block.               

image

[https://github.com/LeonLinAtAMI/libspdm/blob/58d36b18c92c46d8533e6f61a4facaad86138464/library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c#L183](https://github.com/LeonLinAtAMI/libspdm/blob/58d36b18c92c46d8533e6f61a4facaad86138464/library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c#L183)
 
 I found the current guarding check inside validate_responder_capability function is not set for this case for multi_key_cap.
 
image

[https://github.com/DMTF/libspdm/blob/5f3d9294e377c53df8c956603be8b153a706b2d3/library/spdm_requester_lib/libspdm_req_get_capabilities.c#L167](https://github.com/DMTF/libspdm/blob/5f3d9294e377c53df8c956603be8b153a706b2d3/library/spdm_requester_lib/libspdm_req_get_capabilities.c#L167)

@steven-bellock 
So is your idea that removing the default block and then adding guarding check into validate_responder_capability?

@jyao1 
I have probably not taken all edge cases into consideration.
Please let me know if I missed something or misunderstand on this.

Comment (steven-bellock) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:183]: >I found the current guarding check inside validate_responder_capability function is not set for this case for multi_key_cap.

The `default` branch cannot be taken if `validate_responder_capability` returns `true`.

Comment (LeonLinAtAMI) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:183]: Yes, I got your point. And removed default block and added new unit test for the case in GET_CAPABILITIES.

File: library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c [hunk 2]
```diff
@@ -180,8 +180,6 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
                 spdm_context->connection_info.multi_key_conn_rsp = true;
             }
             break;
-        default:
-            return LIBSPDM_STATUS_INVALID_MSG_FIELD;
```
Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:184]: This is defensive programming.
Please keep it.

Comment (LeonLinAtAMI) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:184]: The original memory leak is found by unit testing. And I added this line intuitively. Also there is an unit test for the case.
image

I just resolved the @steven-bellock review that removing default block and related unit test. I ensured that original function of validate_responder_capability had already defended the case of mul_key_info_cap 11b. Also added new unit test in in [get_capabilities.c](https://github.com/DMTF/libspdm/pull/3736/changes#diff-5bd489a48e149f80170d816048a2cde5749f16986280e581d72b1981d51818a5R2656).

@jyao1  Is you idea keeping the original codes or the changed with libspdm_release_sender_buffer beforehand.

I probably rarely considered all edge cases.

Please let me know if your defensive programming guarded some unexpected status inside SPDM context or unexpected order during SPDM VCA and the modifications out of scope of libspdm_req_negotiate_algorithms.c for examples: SPDM base specification, and libspdm_req_get_capabilities.c.

It is okay to me reverting default block and specific unit test for it.

Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:184]: If it is impossible, then we should add ASSERT (FALSE) to ensure it.

Comment (LeonLinAtAMI) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:184]: @jyao1 I updated the codes according to your suggestions. Please let me know if any enhancement. Thanks.

File: unit_test/test_spdm_requester/get_capabilities.c [hunk 3]
```diff
@@ -1073,380 +1131,1545 @@ static libspdm_return_t receive_message(
...
+    status = libspdm_get_capabilities(spdm_context);
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(spdm_context->connection_info.capability.ct_exponent, 0);
+    assert_int_equal(
+        spdm_context->connection_info.capability.flags,
+        !(SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CACHE_CAP |
+          SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CERT_CAP |
+          SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHAL_CAP |
+          SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP_SIG |
+          SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_FRESH_CAP));
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/get_capabilities.c:1810]: The `!` should probably be a `~`?

File: unit_test/test_spdm_requester/get_capabilities.c [hunk 4]
```diff
@@ -1073,380 +1131,1545 @@ static libspdm_return_t receive_message(
...
+    }
+}
+
+/*
+ * static void req_get_capabilities_case1(void **state)
+ * {
+ * }
+ */
+
+static void req_get_capabilities_case2(void **state)
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/get_capabilities.c:1672]: Please document the test cases as shown in https://github.com/DMTF/libspdm/blob/main/doc/internal/unit_test_template.md.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: unit_test/test_spdm_requester/get_capabilities.c (hunk 3 above)
Comment (LeonLinAtAMI) [reply unit_test/test_spdm_requester/get_capabilities.c:1810]: I did not add these lines of codes.
I thought git diff tool here confused not only reviewers but also me.

image

From another perspective of diff tool like this below. 
https://www.diffchecker.com/oZZb7ni3/

File: unit_test/test_spdm_requester/get_capabilities.c (hunk 4 above)
Comment (LeonLinAtAMI) [reply unit_test/test_spdm_requester/get_capabilities.c:1672]: @steven-bellock  Thanks for your review. 

I did not add these lines of codes.
The git diff tool here confused not only reviewers but also me.

The new test cases added by me had been compliant with the doc you mentioned above.
image

From another perspective of diff tool like this below. 
https://www.diffchecker.com/oZZb7ni3/

Please let me know if something done wrong. Thanks a lot.

**Comment by LeonLinAtAMI** [comment]
The git diff tool here confused not only reviewers but also me.
I used built-in diff tool of source control inside vscode instead.

From another perspective of diff tool like this below.
https://www.diffchecker.com/oZZb7ni3/

**Review by steven-bellock — APPROVED** [review]
